lms/djangoapps/teams/static/teams/js/views/edit_team.js

```diff
@@ -0,0 +1,164 @@
+
+                   this.teamDescriptionField = new FieldViews.TextareaFieldView({
+                       model: new TeamModel(),
+                       title: gettext("Team Description (Required) *"),
+                       valueAttribute: 'description',
+                       editable: 'always',
+                       showMessages: false,
+                       descriptionMessage: gettext("A short description of the team to help other learners understand the goals or direction of the team (maximum 300 characters).")
+                   });
+
+                   this.optionalDescriptionField = new FieldViews.ReadonlyFieldView({
+                       model: new TeamModel(),
+                       title: gettext("Optional Characteristics"),
+                       valueAttribute: 'optional_description',
+                       helpMessage: gettext("Help other learners decide whether to join your team by specifying some characteristics for your team. Choose carefully, because fewer people might be interested in joining your team if it seems too restrictive. You cannot change these characteristics after you create the team.")
+                   });
+
+                   this.teamLanguageField = new FieldViews.DropdownFieldView({
+                       model: new TeamModel(),
+                       title: gettext("Language"),
+                       valueAttribute: 'language',
+                       required: false,
+                       showMessages: false,
+                       titleIconName: 'fa-comment-o',
+                       options: this.languages,
+                       helpMessage: gettext("The language that team members primarily use to communicate with each other.")
+                   });
+
+                   this.teamCountryField = new FieldViews.DropdownFieldView({
+                       model: new TeamModel(),
+                       title: gettext('Country'),
+                       valueAttribute: 'country',
+                       required: false,
+                       showMessages: false,
+                       titleIconName: 'fa-globe',
+                       options: this.countries,
+                       helpMessage: gettext("The country that team members primarily identify with.")
+                   });
+               },
+
+               render: function() {
+                   this.$el.html(_.template(edit_team_template)({}));
+                   this.set(this.teamNameField, '.team-required-fields');
+                   this.set(this.teamDescriptionField, '.team-required-fields');
+                   this.set(this.optionalDescriptionField, '.team-optional-fields');
+                   this.set(this.teamLanguageField, '.team-optional-fields');
+                   this.set(this.teamCountryField, '.team-optional-fields');
+                   return this;
+               },
+
+               set: function(view, selector) {
+                   this.$el.find(selector).append(view.$el);
```

Comment:
Ah, I see @muhammad-ammar. The problem with using `jQuery.append` is that it prevents the `render` function from being able to be called many times.

As a compromise, you can check if the view has already been rendered. If it has, re-render the view and call `setElement` on the view's el. Otherwise if it hasn't been rendered yet, append.

```javascript
var viewEl = view.$el;
if (this.$(selector).contains(viewEl)) {
    view.render().setElement(viewEl);
} else {
    this.$(selector).append(view.render().$el);
}
```